zuul-core/src/test/java/com/netflix/netty/common/CloseOnIdleStateHandlerTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License");
+ *      you may not use this file except in compliance with the License.
+ *      You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *      Unless required by applicable law or agreed to in writing, software
+ *      distributed under the License is distributed on an "AS IS" BASIS,
+ *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *      See the License for the specific language governing permissions and
+ *      limitations under the License.
+ */
+
+package com.netflix.netty.common;
+
+import static io.netty.handler.timeout.IdleStateEvent.ALL_IDLE_STATE_EVENT;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import com.netflix.spectator.api.Counter;
+import com.netflix.spectator.api.Registry;
+import com.netflix.zuul.netty.server.http2.DummyChannelHandler;
+import io.netty.channel.embedded.EmbeddedChannel;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CloseOnIdleStateHandlerTest {
+
+    @Mock
+    private Registry registry;
```

Comment:
Consider using a default registry and checking the metric value at the end of the test

Reply:
yep, good call.

Reply:
actually, I don't see an API to lookup the counter for `DefaultRegistry`. 
(I'd have to inject the counter obj itself then, which isn't clean.)